### PR TITLE
AutoEnsemble & XGB models

### PR DESCRIPTION
**Pull request by gavinkatz001**
Added yaml files for 5 and 15 min intervals. Would like some guidance from the rest of the team on what values to play around with and what to put for them (ie what combination of forecasters we should pass to AutoEnsemble).

### CODE REVIEW

**Comment by Tony911029** [comment]
@gavinkatz001 Hey Gavin, looks good! Have you tried it on 3.10 Notebook?

```
## Change your yaml path here
yaml_path = "../../src/tuning/configs/1_exponential_smooth_15min.yaml"
```
You can point to your yaml. Just want to make sure things work. If everything is fine. We can do some runs just to see how it performance.

**Comment by gavinkatz001** [comment]
> @gavinkatz001 Hey Gavin, looks good! Have you tried it on 3.10 Notebook?
> 
> ```
> ## Change your yaml path here
> yaml_path = "../../src/tuning/configs/1_exponential_smooth_15min.yaml"
> ```
> 
> You can point to your yaml. Just want to make sure things work. If everything is fine, we can do some runs just to see how it performs.

Not yet, I will give it a try now 👍

**Comment by gavinkatz001** [comment]
image

Stopped it after 8 mins. Not sure how long it would have taken but it didnt throw any errors so I think (and hope) it should be fine.

**Comment by Tony911029** [comment]
> image
> 
> Stopped it after 8 mins. Not sure how long it would have taken but it didnt throw any errors so I think (and hope) it should be fine.

yeah

**Review by Tony911029 — APPROVED** [review]
Goods good to me. @RobotPsychologist Might wanna review the params.

**Comment by gavinkatz001** [comment]
> Goods good to me. @RobotPsychologist Might wanna review the params.

Agreed. Some guidance on this would be helpful

File: src/tuning/configs/8_ensemble_gk_05min.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      ["AutoARIMA", {}],
+       ["ExponentialSmoothing", {}], # not sure about the format for this since AutoEnsembleForecaster expects a (str, forecaster) tuple for the forecasters parameter
+       ["ThetaForecaster", {}]
+    ]
+  method:
+    type: list
+    values: ["feature-importance", "inverse-variance"]
+  test_size:
+    type: list
+    values: [0.1, 0.15, 0.2, 0.25, 0.3]
```
Comment (RobotPsychologist) [inline src/tuning/configs/8_ensemble_gk_05min.yaml:17]: Just pick one value for this, we don't need to hyperparmeter tune how big the internal test set is for this. The reason why, is because this is essentially setting the train/test split, we are going to evaluate again on the 20 day hold out set not in the training data. So what AutoEnsembleForecaster is doing, is running a bunch of these models on an "internal" training set and then comparing all of the internal auto explored sets of models against an "internal" test set. So something like 0.2, or 0.25 is fine.

File: src/tuning/configs/8_ensemble_gk_05min.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+AutoEnsembleForecaster:
+  forecasters:
+    type: list
+
+    # this is just an example. we can put whatever forecasters we want to combine here
+    # the AutoEnsembleForecaster finds the optimal weights for each forecaster so we dont need to tune any parameters ourselves
+    values: [
+      ["AutoARIMA", {}],
```
Comment (RobotPsychologist) [inline src/tuning/configs/8_ensemble_gk_05min.yaml:8]: You should copy the parameters from the other config files for including in the {} I think?

Not sure what the defaults will be if you leave this blank, it could be huge and really slow down the modelling runs. Add things like Naive as well. Maybe TinyTimeMixer.

File: src/tuning/configs/8_ensemble_gk_15min.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    range: [0, 30]
+  lags_future_covariates:
+    type: int
+    range: [0, 30]
+  likelihood:
+    type: list
+    values: ["poisson", "quantile"]
+  quantiles:
+    type: int
+    range: [1,30]
```
Comment (RobotPsychologist) [inline src/tuning/configs/8_ensemble_gk_15min.yaml:37]: This shouldn't be a range, so this is setting up quantile forecasting, essnetially you're saying fit to the 1st, 2nd, 3rd,.... 30th quantiles. If anything this should be something like a list, and make it 95, 90, 75, 50, 25, 10, 5

**Review by RobotPsychologist — CHANGES_REQUESTED** [review]
Great work Gavin, we will need to add a lot more ensemble models, but this is a good start, just a few changes to some parameters that will allow for some sensible initial testing.

**Comment by gavinkatz001** [comment]
will do. thanks for the feedback @RobotPsychologist!

**Review by RobotPsychologist — APPROVED** [review]
LGTM
